tests/test_helpers/test_tenants.py

```diff
@@ -287,3 +288,47 @@ def test_fix_lms_base_port(lms_root_port, domain_name, expected_result):
 def test_fix_lms_base_empty_domain_name(domain_name):
     """Verify that fix_lms_base sets the correct port for the result."""
     assert tenants.fix_lms_base(domain_name) == ''
+
+
+@pytest.mark.django_db
+def test_get_fx_dashboard_url():
+    """Verify _get_fx_dashboard_url reurns correct address"""
+    factory = RequestFactory()
+    request = factory.get('/some-path')
+    request.site = type('Site', (), {})()
+    request.site.name = 's2.sample.com'
+    request.LANGUAGE_CODE = 'en'
+    # User with current site tenant access
+    request.user = get_user_model().objects.get(id=4)
+    url = tenants.get_fx_dashboard_url(request)
+    expected_url = 'http://dashboard.example.com/en/2'
+    assert url == expected_url
+    # User with another tenant access
+    request.user = get_user_model().objects.get(id=3)
+    url = tenants.get_fx_dashboard_url(request)
+    assert url is None
+    # user without any access
+    request.user = get_user_model().objects.get(id=5)
+    url = tenants.get_fx_dashboard_url(request)
+    assert url is None
+
+
+@pytest.mark.django_db
+@pytest.mark.parametrize('lang, expected_lang', [
+    ('en', 'en'),
+    ('ar', 'ar'),
+    ('fr', 'en'),
+    ('', 'en'),
+    (None, 'en'),
+])
+def test_get_fx_dashboard_url_for_language(lang, expected_lang):
+    """Verify _get_fx_dashboard_url for default language code"""
+    factory = RequestFactory()
+    request = factory.get('/some-path')
+    request.site = type('Site', (), {})()
+    request.site.name = 's2.sample.com'
```

Comment:
Perhaps use Site.objects.get() to test with an actual site from the `base_data`.